### PR TITLE
[WIP] Add tests to `getJSONObject` for queries involving single quotes

### PR DESCRIPTION
**Pull request by SurajAralihalli**
Depends on [PR 15082](https://github.com/rapidsai/cudf/pull/15082)

Cudf [PR 15082](https://github.com/rapidsai/cudf/pull/15082) fixes issues with `get_json_object` when single quotes are used in the JSONPath query. Additionally, it now throws a `std::invalid_argument` error for invalid queries. The JNI function catches this exception and produces a string column consisting of null values.

This PR verifies the output of `get_json_object` between the cudf implementation and the standard Spark version.

For example in `{ "A.B": 2, "'A": { "B'": 3 } }`

| JSONPath      | Before CUDF Fix                                            | Spark        | After Fix           |
|---------------|-------------------------------------------------------|----------------------|---------------------|
| $.'A.B'       | 2                                                     | 3                    | 3                   |
| $.'A          | CUDF_FAIL("Encountered invalid JSONPath input string")| {"B'": 3}            | {"B'": 3}           |
| ${A}        | CUDF_FAIL("Unrecognized JSONPath operator")| null            | null           |

### CODE REVIEW

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuGetJsonObject.scala [hunk 1]
```diff
@@ -33,8 +33,14 @@ case class GpuGetJsonObject(json: Expression, path: Expression)
...
 
   override def doColumnar(lhs: GpuColumnVector, rhs: GpuScalar): ColumnVector = {    
-    lhs.getBase().getJSONObject(rhs.getBase, 
-    GetJsonObjectOptions.builder().allowSingleQuotes(true).build());
+    try {
+        lhs.getBase().getJSONObject(rhs.getBase, 
+            GetJsonObjectOptions.builder().allowSingleQuotes(true).build());
+    }
+    catch {
+        case _: CudfException => GpuColumnVector.columnVectorFromNull(
```
Comment (SurajAralihalli) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuGetJsonObject.scala:41]:
```
ai.rapids.cudf.CudfException: CUDF failure at:/home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-682-cuda11/thirdparty/cudf/cpp/src/json/json_path.cu:595: Unrecognized JSONPath operator
E                       at ai.rapids.cudf.ColumnView.getJSONObject(Native Method)
E                       at ai.rapids.cudf.ColumnView.getJSONObject(ColumnView.java:2996)
E                       at com.nvidia.spark.rapids.GpuGetJsonObject.doColumnar(GpuGetJsonObject.scala:37)
```

**Comment by revans2** [comment]
How is this different from https://github.com/NVIDIA/spark-rapids/pull/10466? If we are fixing some things in CUDF, that is great, but I think the other PR has duplicated a lot of the work that would be needed here.

**Comment by SurajAralihalli** [comment]
> How is this different from #10466? If we are fixing some things in CUDF, that is great, but I think the other PR has duplicated a lot of the work that would be needed here.

Thanks for pointing it out! I have limited the scope of this PR to only add tests for the valid queries that include single quotes and currently fail due to the bugs in cudf. I believe it is not covered in the scope of https://github.com/NVIDIA/spark-rapids/pull/10466 as it examines invalid queries.

**Comment by revans2** [comment]
Sounds good thanks for adding these tests.

File: integration_tests/src/main/python/get_json_test.py [hunk 2]
```diff
@@ -66,6 +66,20 @@ def test_get_json_object_single_quotes():
...
+
+    assert_gpu_and_cpu_are_equal_collect(
+        lambda spark: spark.createDataFrame(data,schema=schema).select(
+        f.get_json_object('jsonStr',"$.AB").alias('sub_a'),
+        f.get_json_object('jsonStr',"$['A.B']").alias('sub_b'),
+        f.get_json_object('jsonStr',"$.'A.B'").alias('sub_c'),
+        f.get_json_object('jsonStr',"$.A.B").alias('sub_d'),
+        f.get_json_object('jsonStr',"$.'A").alias('sub_e')
+        ),
+        conf={'spark.rapids.sql.expression.GetJsonObject': 'true'})
```
Comment (thirtiseven) [inline integration_tests/src/main/python/get_json_test.py:81]: nit: This config is now true by default so it's safe to remove it here.

**Review by thirtiseven — APPROVED** [review]
It is not a cudf_dependency now. Will be good to add this case.

**Pull request closed without merging**